docs/cpp/cpp-refactoring.md

```diff
@@ -0,0 +1,73 @@
+
+![Create a definition and a declaration in the same file](images/refactoring/create-declaration-and-definition-same-file.gif)
+
+If you defined declarations or definitions in a different file from your source file, the extension follows your convention. For example, for a header and a source file with matching names, both are identified, even if the header file has not been included in the source file. In this case, once the declaration or definition is added, your header file is automatically referenced in your source file for you.
+
+If you do not have any precedent of where you create definitions or declarations, we will create a header or source file with a matching name to your current file for you.
+
+For templates, if a function template is declared in a header file, the definition of that function template will be created in the same header file. This also applies for nontemplate member functions of class templates.
+
+### Copy declarations or definitions
+
+If you want to choose the location that your declaration or definition is added in code, you can use the code action **Copy Declaration/Definition**. This adds the declaration or definition to your clipboard rather than directly to your code.
+
+To invoke the code action, select a function that has a Quick Fix available, then select the Code Action (light bulb) and choose **Copy definition of ‘YourFunctionName’** or **Copy declaration of ‘YourFunctionName’**.
+
+![Copy a declaration or definition](images/refactoring/copy-declaration-definition.gif)
+
+If the declaration/definition is not formatted when you paste, turn on auto formatting using **Settings** (`kb(workbench.action.openSettings)`) > **Format On Paste**. Enabling this will automatically format all items you paste in the editor.
+
+## Extract to method
+
+The Extract Method refactoring feature allows you to extract a block of code into a separate method to help improve code readability, reduce duplication, and make the code more modular.
+
+To extract a method, select the C++ code you would like to extract. A code action (light bulb) will appear with the option **Extract to Function**. Otherwise, right-click on the code and select **Refactor > Extract** or use the keyboard command (`Ctrl + Shift + R, Ctrl + E` on windows) to get more information. You can then name the new function that has been created. The new function containing your highlighted code will be placed above the current function.
+
+![Extract Method and create declaration](images/refactoring/extract-method.gif)
+
+## Quick Fixes/Code Actions
+
+The C/C++ Extension provides C/C++ specific suggestions for how to fix and improve your C++ code. These can be accessed through a code action (lightbulb icon) as you code or through the Quick Fixes link when you hover over an item. These are available for many scenarios discussed previously, such as extract to method, copy declaration/definition, or create declaration/definition. Otherwise, a code actions can help:
```

Comment:
```suggestion
The C/C++ extension provides C/C++ specific suggestions for how to fix and improve your C++ code. These can be accessed through a code action (lightbulb icon) as you code or through the Quick Fixes link when you hover over an item. These are available for several scenarios, such as extract to method, copy declaration/definition, or create declaration/definition. Otherwise, a code actions can help:
```

Reply:
What are you referring to by "These can be accessed"? "These suggestions"? Can you be more specific?
Also, the last sentence seems to end unexpectedly.

Reply:
The goal with this paragraph was to show that we make suggestions to the user based on context, and that this can be shown to a user in two different ways: a quick fix or a code action. However, it is important to know what these are the same thing under the hood, just two different ways of accessing the suggestion information.